x-pack/legacy/plugins/epm/public/screens/detail/header.tsx

```diff
@@ -3,52 +3,76 @@
  */
-
-import React, { Fragment, useCallback, useState } from 'react';
-import {
-  EuiButton,
-  EuiButtonEmpty,
-  EuiButtonEmptyProps,
-  EuiFlexGroup,
-  EuiFlexItem,
-  EuiPage,
-  EuiTitle,
-  IconType,
-} from '@elastic/eui';
+import React, { Fragment, useCallback, useState, useMemo } from 'react';
 import styled from 'styled-components';
-// TODO: either
-// * import from a shared point
-// * duplicate inside our folder
-import { useTrackedPromise } from '../../../../infra/public/utils/use_tracked_promise';
-import { installPackage } from '../../data';
+import { EuiFlexGroup, EuiFlexItem, EuiPage, EuiTitle, IconType } from '@elastic/eui';
 import { PLUGIN } from '../../../common/constants';
 import { PackageInfo } from '../../../common/types';
 import { Version } from '../../components/version';
 import { IconPanel } from '../../components/icon_panel';
-import { useBreadcrumbs, useCore, useLinks } from '../../hooks';
+import { useBreadcrumbs, useLinks } from '../../hooks';
 import { CenterColumn, LeftColumn, RightColumn } from './layout';
+import { InstallationButton } from './installation_button';
+import { ConfirmPackageInstall } from './confirm_package_install';
+// TODO: either
+// * import from a shared point
+// * duplicate inside our folder
+import { useTrackedPromise } from '../../../../infra/public/utils/use_tracked_promise';
+import { installPackage } from '../../data';
+import { NavButtonBack } from './nav_button_back';
+
+const FullWidthNavRow = styled(EuiPage)`
+  /* no left padding so link is against column left edge  */
+  padding-left: 0;
+`;
+
+const Text = styled.span`
+  margin-right: ${props => props.theme.eui.euiSizeM};
+`;
+
+const StyledVersion = styled(Version)`
+  font-size: ${props => props.theme.eui.euiFontSizeS};
+  color: ${props => props.theme.eui.euiColorDarkShade};
+`;
 
 type HeaderProps = PackageInfo & { iconType?: IconType };
 
 export function Header(props: HeaderProps) {
-  const { iconType, title, version } = props;
-  const { theme } = useCore();
+  const [isModalVisible, setModalVisible] = useState<boolean>(false);
+  const { iconType, title, version, assets } = props;
   const { toListView } = useLinks();
   useBreadcrumbs([{ text: PLUGIN.TITLE, href: toListView() }, { text: title }]);
+  const [isInstalled, setInstalled] = useState<boolean>(props.status === 'installed');
+  const [installationRequest, attemptInstallation] = useTrackedPromise(
+    {
+      createPromise: async () => {
+        const pkgkey = `${props.name}-${props.version}`;
+        return installPackage(pkgkey);
+      },
+      onResolve: (value: PackageInfo) => setInstalled(value.status === 'installed'),
+    },
+    [props.name, props.version, installPackage]
+  );
+  const isLoading = installationRequest.state === 'pending';
 
-  const FullWidthNavRow = styled(EuiPage)`
-    /* no left padding so link is against column left edge  */
-    padding-left: 0;
-  `;
+  const handleModal = useCallback(() => {
```

Comment:
`toggleModal` is clearer to me. I find `handle` a good prefix for an action type. eg. `handleClick`, `handleCancel`.